api/src/test/java/org/openmrs/scheduler/tasks/AutoRetireUsersTaskTest.java

```diff
@@ -0,0 +1,161 @@
+		userService.setUserProperty(
+			user,
+			OpenmrsConstants.USER_PROPERTY_LAST_LOGIN_TIMESTAMP,
+			String.valueOf((System.currentTimeMillis() - TWO_DAYS_IN_MILLISECONDS))
+		);
+
+		user.getAllRoles().forEach(user::removeRole);
+
+		autoRetireUsersTask.execute();
+
+		assertTrue(user.isRetired());
+		assertEquals(user.getRetireReason(), AUTO_RETIRE_REASON);
+	}
+
+	@Test
+	public void shouldNotRetireUsersWhoseInactivityDoNotExceedNumberOfDaysToRetire() {
+		administrationService.saveGlobalProperty(new GlobalProperty(OpenmrsConstants.GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS, TWO_DAYS_PROPERTY_VALUE));
+
+		User user = getDefaultUser();
+		userService.setUserProperty(
+			user,
+			OpenmrsConstants.USER_PROPERTY_LAST_LOGIN_TIMESTAMP,
+			String.valueOf((System.currentTimeMillis() - ONE_DAY_IN_MILLISECONDS))
+		);
+
+		user.getAllRoles().forEach(user::removeRole);
+
+		autoRetireUsersTask.execute();
+
+		assertFalse(user.isRetired());
+	}
+
+	@Test
+	public void shouldNotRetireAlreadyRetiredUsers() {
+		administrationService.saveGlobalProperty(new GlobalProperty(OpenmrsConstants.GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS, ONE_DAY_PROPERTY_VALUE));
+
+		String retireReason = "Retire Test users";
+
+		User retiredUser = userService.getUser(1);
+		userService.setUserProperty(
+			retiredUser,
+			OpenmrsConstants.USER_PROPERTY_LAST_LOGIN_TIMESTAMP,
+			String.valueOf((System.currentTimeMillis() - TWO_DAYS_IN_MILLISECONDS))
+		);
+
+		retiredUser.getAllRoles().forEach(retiredUser::removeRole);
+
+		retiredUser.setRetired(true);
+		retiredUser.setRetireReason(retireReason);
+		
+		userService.saveUser(retiredUser);
+
+		autoRetireUsersTask.execute();
+
+		User fetchedUser = userService.getUser(retiredUser.getUserId());
+		assertTrue(fetchedUser.isRetired(), "User should remain retired after the task runs");
+		assertEquals(fetchedUser.getRetireReason(), retireReason);
+	}
+
+	@Test
+	public void shouldNotRetireUsersThatAreInactiveSinceCreationAndInactivityDoesNotExceedNumberOfDaysToRetire() {
+		administrationService.saveGlobalProperty(new GlobalProperty(OpenmrsConstants.GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS, ONE_DAY_PROPERTY_VALUE));
+
+		User user = getDefaultUser();
+
+		user.getAllRoles().forEach(user::removeRole);
+
+		user.setDateCreated(new Date(System.currentTimeMillis() - TWENTY_THREE_HOURS_IN_MILLISECONDS));
+
+		autoRetireUsersTask.execute();
+
+		assertFalse(user.isRetired());
+	}
+
+	@Test
+	public void shouldRetireUsersThatAreInactiveSinceCreationAndInactivityExceedsNumberOfDaysToRetire() {
+		administrationService.saveGlobalProperty(new GlobalProperty(OpenmrsConstants.GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS, ONE_DAY_PROPERTY_VALUE));
+
+		User user = getDefaultUser();
+
+		user.getAllRoles().forEach(user::removeRole);
+
+		user.setDateCreated(new Date(System.currentTimeMillis() - TWO_DAYS_IN_MILLISECONDS));
+
+		autoRetireUsersTask.execute();
+
+		assertTrue(user.isRetired());
+		assertEquals(user.getRetireReason(), AUTO_RETIRE_REASON);
```

Comment:
Of the two arguments, which is the expected? `AUTO_RETIRE_REASON` or `user.getRetireReason()`?

Reply:
`AUTO_RETIRE_REASON` is the expected. I will switch them around.